tick/eval_test.go

```diff
@@ -266,3 +266,109 @@ var s2 = a.structB()
 		t.Fatal("expected error from Evaluate")
 	}
 }
+
+type structLambda struct {
+	Test tick.Node
+}
+
+func Test_EvalBool(t *testing.T) {
+	script := `a.test(lambda: "value" > 20)`
+
+	scope := tick.NewScope()
+	a := &structLambda{}
+	scope.Set("a", a)
+
+	err := tick.Evaluate(script, scope)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	se := tick.NewStatefulExpr(a.Test)
+
+	executionScope := tick.NewScope()
+	executionScope.Set("value", int64(40))
+	b, err := se.EvalBool(executionScope)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Result: %v", b)
+}
+
+func Benchmark_EvalBool_1ScopeItem(b *testing.B) {
+	script := `a.test(lambda: "value" > 20)`
+
+	scope := tick.NewScope()
+	a := &structLambda{}
+	scope.Set("a", a)
+
+	err := tick.Evaluate(script, scope)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	se := tick.NewStatefulExpr(a.Test)
```

Comment:
Instead of parsing the script and grabing the tick.Node you can create it directly like this:

```
n := &BinaryNode{
    Operator: TokenGreater,
    Left: &ReferenceNode{
        Reference: "value",
   },
    Right: &NumberNode{
       IsInt: true,
       Int64: 20,
   }
}

se := tick.NewStatefulExpr(n)
```